osu.Game/Online/Metadata/OnlineMetadataClient.cs

```diff
@@ -215,8 +218,8 @@ public override async Task EndWatchingUserPresence()
                 if (connector?.IsConnected.Value != true)
                     throw new OperationCanceledException();
 
-                // must happen synchronously before any remote calls to avoid misordering.
-                userStates.Clear();
+                // must happen synchronously before any remote calls to avoid mis-ordering.
```

Comment:
The schedule is likely going to change the execution order, but I'm not too sure about this comment and its implications. @bdach could you please check over this?

Reply:
Also arguably, remaining sets of `isWatchingUserPresence.Value` should be scheduled, but it's a bit awkward structurally. I don't think it will affect behaviour but interested in thoughts.

Reply:
I believe that the comment was referring to the fact that the clear cannot happen in a continuation to avoid something along the lines of the following:

1. `EndWatchingUserPresence()`
2. `StartWatchingUserPresence()`
3. acknowledgement of (1) is received - clear is scheduled via continuation / sync context
4. server receives (2) and starts sending refreshed user presence data
5. schedule from (3) executes and clears the received data, leading in an incorrect final state

Which is related to your other question; I'd say if we're going to be using schedules as our synchronisation primitive, we should be using it across the board, i.e. every time either `userStates` or `isWatchingUserPresence` are touched. I'm not entirely sure what structural awkwardness you were referring to specifically there, it seems like it should be fairly straightforward.

Reply:
Yeah if that's enough then that's fine. It was more whether the `Invoke` call of `EndWatchingUserPresence` also needed to be scheduled (which comes with `async` concerns).